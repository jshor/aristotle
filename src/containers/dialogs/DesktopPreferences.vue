<template>
  <dialog-container
    :dialog-type="ViewType.Preferences"
    title="Preferences"
  >
    <desktop-preferences-view>
      <desktop-preferences-subview title="Colors">
        <property-form
          v-model="colors"
          id="section1"
        />
      </desktop-preferences-subview>
      <desktop-preferences-subview title="Grid">
        <property-form
          v-model="grid"
          id="section1"
        />
      </desktop-preferences-subview>
      <desktop-preferences-subview title="Snapping">
        <property-form
          v-model="snapping"
          id="section1"
        />
      </desktop-preferences-subview>
      <desktop-preferences-subview title="Simulation">
        <property-form
          v-model="simulation"
          id="section1"
        />
      </desktop-preferences-subview>
      <desktop-preferences-subview title="Experience">
        <property-form
          v-model="experience"
          id="section1"
        />
      </desktop-preferences-subview>
    </desktop-preferences-view>
  </dialog-container>
</template>

<script lang="ts" setup>
import { storeToRefs } from 'pinia'
import DesktopPreferencesSubview from '@/components/desktop/DesktopPreferencesSubview.vue'
import DesktopPreferencesView from '@/components/desktop/DesktopPreferencesView.vue'
import DialogContainer from './DialogContainer.vue'
import PropertyForm from '@/components/properties/PropertyForm.vue'
import { usePreferencesStore } from '@/store/preferences'
import { ViewType } from '@/types/enums/ViewType'

const preferencesStore = usePreferencesStore()

const {
  colors,
  grid,
  snapping,
  simulation,
  experience
} = storeToRefs(preferencesStore)
</script>
