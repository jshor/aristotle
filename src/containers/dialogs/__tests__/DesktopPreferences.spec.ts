import { shallowMount } from '@vue/test-utils'
import { setActivePinia, createPinia } from 'pinia'
import DesktopPreferences from '../DesktopPreferences.vue'

setActivePinia(createPinia())

describe('Desktop Preferences Dialog', () => {
  it('should match the snapshot', () => {
    expect(shallowMount(DesktopPreferences).html()).toMatchSnapshot()
  })
})
