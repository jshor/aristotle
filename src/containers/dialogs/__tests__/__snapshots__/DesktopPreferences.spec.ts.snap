// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Desktop Preferences Dialog > should match the snapshot 1`] = `"<dialog-container-stub dialogtype="2" title="Preferences"></dialog-container-stub>"`;

exports[`Desktop Preferences Dialog should match the snapshot 1`] = `"<dialog-container-stub dialogtype="2" title="Preferences"></dialog-container-stub>"`;
